torchrl/modules/llm/vllm_policy.py

```diff
@@ -280,10 +281,18 @@ def get_output_tokens_and_log_probs(td, padding_value=padding_value):
                     layout=torch.strided
                 ).to_padded_tensor(padding=padding_value)
             tokens_response_td.rename_key_("token_ids", "tokens_response")
-            # td["tokens_response"] = outputs.token_ids
             if return_log_probs:
+                padded_values = tokens_response_td["tokens_response"] == padding_value
                 tokens_response_td.rename_key_("logprobs", "log_probs")
-                # td["log_probs"] = outputs.logprobs.unsqueeze(-1)
+                if padded_values.any():
+                    print(
+                        "padded_values:",
+                        padded_values.sum(),
+                        torch.where(padded_values),
+                    )
+                    lps = tokens_response_td["log_probs"]
+                    lps = torch.where(expand_as_right(~padded_values, lps), lps, 0.0)
```

Comment:
should this be 1.0?

Reply:
To me 1 makes more sense, it's the log prob of prob = 1 (you don't do the ratio of log-probs but their difference).
Any value would do, but I don't see why choosing p=2.71 is superior to 1